lib/matplotlib/tri/triangulation.py

```diff
@@ -80,18 +80,15 @@ def __init__(self, x, y, triangles=None, mask=None):
                     f'range 0 <= i < {len(self.x)} but found value '
                     f'{self.triangles.min()}')
 
-        if mask is not None:
-            self.mask = np.asarray(mask, dtype=bool)
-            if self.mask.shape != (self.triangles.shape[0],):
-                raise ValueError('mask array must have same length as '
-                                 'triangles array')
-
         # Underlying C++ object is not created until first needed.
         self._cpp_triangulation = None
 
         # Default TriFinder not created until needed.
         self._trifinder = None
 
+        # Set mask
+        self.set_mask(mask)
```

Comment:
This initializes `self._trifinder` as a side effect? I'm not sure if that is wanted/needed. Three lines above we state that we want to defer this initialization.
If the proposed code is what we want. the comment above should be changed to "initialized via set_mask()".

Reply:
No, since `_trifinder = None` here (which is why I moved the set_mask call from where I removed the "duplicate" code.

https://github.com/matplotlib/matplotlib/blob/29a86636a9c45ab5ac4d80ac76eaee497f460dce/lib/matplotlib/tri/triangulation.py#L243-L244

Reply:
So the only redundancy is that _edges and _neighbours are set to None twice.

Reply:
Oh, I overlooked the `not` 🙄